### ISSUE TITLE
Route cache exception on Laravel 5.4

### ISSUE THREAD

**Issue by dyzajash** [spatie/laravel-feed#22]
Hello, 

I've just upgraded my CMS to Laravel 5.4 and when I try to do 

```
➜  joseph-cms git:(develop) ✗ php artisan route:cache
Route cache cleared!
Routes cached successfully!
```

My app just got strange exception: 

```
[2017-01-25 19:00:50] local.ERROR: InvalidArgumentException: Route [spatieLaravelFeed0] not defined. in /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Routing/UrlGenerator.php:304
Stack trace:
#0 /Users/msmolik/Development/joseph-cms/vendor/spatie/laravel-feed/src/FeedServiceProvider.php(47): Illuminate\Routing\UrlGenerator->route('spatieLaravelFe...')
#1 /Users/msmolik/Development/joseph-cms/vendor/spatie/laravel-feed/src/FeedServiceProvider.php(15): Spatie\Feed\FeedServiceProvider->bindFeedLinks()
#2 [internal function]: Spatie\Feed\FeedServiceProvider->boot()
#3 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(28): call_user_func_array(Array, Array)
#4 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Support/helpers.php(912): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#5 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(86): value(Object(Closure))
#6 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(30): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#7 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Container/Container.php(524): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#8 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(762): Illuminate\Container\Container->call(Array)
#9 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(745): Illuminate\Foundation\Application->bootProvider(Object(Spatie\Feed\FeedServiceProvider))
#10 [internal function]: Illuminate\Foundation\Application->Illuminate\Foundation\{closure}(Object(Spatie\Feed\FeedServiceProvider), 25)
#11 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(746): array_walk(Array, Object(Closure))
#12 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/BootProviders.php(17): Illuminate\Foundation\Application->boot()
#13 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(208): Illuminate\Foundation\Bootstrap\BootProviders->bootstrap(Object(Illuminate\Foundation\Application))
#14 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(160): Illuminate\Foundation\Application->bootstrapWith(Array)
#15 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(144): Illuminate\Foundation\Http\Kernel->bootstrap()
#16 /Users/msmolik/Development/joseph-cms/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(116): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#17 /Users/msmolik/Development/joseph-cms/public/index.php(53): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#18 /Users/msmolik/Development/joseph-cms/server.php(21): require_once('/Users/msmolik/...')
#19 {main}  
```

in routes/web.php I've: `Route::feeds();`

And in config/app.php I have: 

```
        /*
         * Application Service Providers...
         */
        App\Providers\AppServiceProvider::class,
        App\Providers\AuthServiceProvider::class,
        // App\Providers\BroadcastServiceProvider::class,
        App\Providers\EventServiceProvider::class,
        App\Providers\RouteServiceProvider::class,

        /*
         * Special Package Service Providers...
         */
        Spatie\Feed\FeedServiceProvider::class,
```

Route cache was working smoothly with laravel-feed on 5.3. 

I was trying to generate clean App and include only this package, with the same results.

**Comment by freekmurze** [comment]
@sebastiandedeyne do you have the same problem at your blog?

**Comment by sebastiandedeyne** [comment]
Are you sure your feed ever worked? Route caching never was an option since this package uses a closure.

You also need to register the feeds in your routes file.

```php
Route::feeds();
```

**Comment by dyzajash** [comment]
> Route caching never was an option since this package uses a closure.

That's strange, because with `Route::feeds()` in web.php, `php artisan route:cache` don't return any errors or warnings. 

For example, when I paste in `web.php`:

```
Route::get('user/{id}', function ($id) {
    return 'User '.$id;
});
```

**route:cache** returns with error about closure usage as you mention.

**Comment by sebastiandedeyne** [comment]
And visiting the feed url works?

Also, what's the output of `route:list`?

**Comment by dyzajash** [comment]
Now I see! 

App with Laravel 5.3 **without** route:cache: **works** as expected. 

App with Laravel 5.3 **with** route:cache: white page. **No errors** in logs + **no warning** when using `php artisan route:cache`

App with Laravel 5.4 **without** route:cache: **works** as expected. 

App with Laravel 5.4 **with** route:cache: **error 500 with data** as I mentioned in 1st post. And also **no warning** when using `php artisan route:cache`. 

I thought it was working with route:cache, because for almost 6 months I don't use `php artisan route:cache` in production... Sorry to bother you.

Can you at least mention that package is not working with `php artisan route:cache` in README.md ?

**Comment by sebastiandedeyne** [comment]
I'm gonna try to refactor the closure to a controller so route caching works 👍

**Comment by sebastiandedeyne** [comment]
Voila, available in `1.2.0`!

https://github.com/spatie/laravel-feed/releases/tag/1.2.0

Gave it a spin on my own site and had no issues.
